.github/scripts/setWalletEnv.ts

```diff
@@ -0,0 +1,71 @@
+// Add some wallet variables to the GITHUB_ENV from a given env name. For example:
+// yarn ts-node .github/scripts/setWalletEnv.ts "alfajores"
+
+import { appendFileSync, existsSync, readFileSync } from 'fs'
+import { resolve } from 'path'
+import { config, exec } from 'shelljs'
+
+config.fatal = true
+config.verbose = true
+
+// Check if the environment argument is provided
+if (process.argv.length < 3) {
+  console.error('Error: You must provide an environment name')
+  process.exit(1)
+}
+
+const envName = process.argv[2]
+
+// Load and export variables from the .env file
+const envFilePath = resolve(`.env.${envName}`)
+if (!existsSync(envFilePath)) {
+  console.error(`Error: Environment file .env.${envName} not found`)
+  process.exit(1)
+}
+
+const GITHUB_ENV = process.env.GITHUB_ENV
+if (!GITHUB_ENV) {
+  console.error('Error: GITHUB_ENV is not set')
+  process.exit(1)
+}
+
+// Append commit hash to GITHUB_ENV
+const commitHash = exec('git rev-parse HEAD', { silent: true }).toString().trim()
+appendFileSync(GITHUB_ENV, `WALLET_COMMIT_HASH=${commitHash}\n`)
+
+// Append commit unix time to GITHUB_ENV
+const commitUnixTime = exec('git show -s --format=%ct', { silent: true }).toString().trim()
+appendFileSync(GITHUB_ENV, `WALLET_COMMIT_UNIX_TIME=${commitUnixTime}\n`)
+
+// Append wallet version to GITHUB_ENV
+const walletVersion = require(resolve('package.json')).version
+appendFileSync(GITHUB_ENV, `WALLET_VERSION=${walletVersion}\n`)
+
+readFileSync(envFilePath, { encoding: 'utf-8' })
+  .toString()
+  .trim()
+  .split('\n')
+  .map((line) => line.trim())
+  .filter((line) => !line.startsWith('#'))
+  .forEach((variable) => {
+    const [key, value] = variable.split('=')
+    if (key && value) {
+      const cleanKey = key.trim()
+      const cleanValue = value
+        .trim()
+        .replace(/^['"]|['"]$/g, '') // Remove surrounding quotes
+        .trim()
+      process.env[cleanKey] = cleanValue
+    }
+  })
```

Comment:
nit: what do you think of using using dotenv instead? From this repo it's easier to use since we already have package.json.
We'd need to move the step that uses this script after `yarn install` but I think this is ok.